kernelci/lab/lava/__init__.py

```diff
@@ -62,16 +62,14 @@ def job_file_name(self, params):
     def _get_priority(self, plan_config):
         # Scale the job priority (from 0-100) within the available levels
         # for the lab, or use the lowest by default.
-        if 'priority' in plan_config.params:
-            priority = plan_config.params['priority']
-            if priority > 100:
-                priority = 100
-        else:
-            priority = 20
-
-        prio_range = self.config._priority_max - self.config._priority_min
-        priority = int(((priority * prio_range) / 100) +
-                       self.config._priority_min)
+        priority = plan_config.params.get('priority', 20)
+        if self.config.priority:
+            priority = int(priority * self.config.priority / 100)
```

Comment:
The intent of the old code was that if a single priority was set then we'd use exactly that priority rather than scaling within it (min and max would end up being set to the same value IIRC). I can't see anyone objecting to us lowering the priority though. The code was written the way it was just to ensure there were no changes for any labs that didn't explicitly configure a range.

Reply:
OK, I guess that's related to some changes in how the YAML config is handled as I've reworked it to be able to dump the input config back into YAML.  That meant removing some mangling, basically it's better if the config objects stay true to the original YAML data and the code using that adds any extra logic on top.

So I'll tweak this again to not scale when there's no range as that's a separate thing, and just include the fix for when the min/max values are not provided as in that case they now resolve as `None` from the config object.